cmd/api/src/api/v2/ad_entity.go

```diff
@@ -0,0 +1,275 @@
+			"controllables":    adAnalysis.FetchOutboundADEntityControl,
+			"controllers":      adAnalysis.FetchInboundADEntityControllers,
+			"gpos":             adAnalysis.FetchEnforcedGPOs,
+		}
+	)
+
+	s.handleAdEntityInfoQuery(response, request, entityType, countQueries)
+}
+
+func (s *Resources) GetContainerEntityInfo(response http.ResponseWriter, request *http.Request) {
+	var (
+		entityType   = ad.Container
+		countQueries = map[string]any{
+			"controllers": adAnalysis.FetchInboundADEntityControllers,
+		}
+	)
+
+	s.handleAdEntityInfoQuery(response, request, entityType, countQueries)
+}
+
+func (s *Resources) GetDomainEntityInfo(response http.ResponseWriter, request *http.Request) {
+	var (
+		entityType   = ad.Domain
+		countQueries = map[string]any{
+			"users":                 adAnalysis.CreateDomainContainedEntityListDelegate(ad.User),
+			"groups":                adAnalysis.CreateDomainContainedEntityListDelegate(ad.Group),
+			"computers":             adAnalysis.CreateDomainContainedEntityListDelegate(ad.Computer),
+			"ous":                   adAnalysis.CreateDomainContainedEntityListDelegate(ad.OU),
+			"gpos":                  adAnalysis.CreateDomainContainedEntityListDelegate(ad.GPO),
+			"foreignUsers":          adAnalysis.CreateForeignEntityMembershipListDelegate(ad.User),
+			"foreignGroups":         adAnalysis.CreateForeignEntityMembershipListDelegate(ad.Group),
+			"foreignGPOControllers": adAnalysis.FetchForeignGPOControllers,
+			"foreignAdmins":         adAnalysis.FetchForeignAdmins,
+			"inboundTrusts":         adAnalysis.CreateDomainTrustListDelegate(graph.DirectionInbound),
+			"outboundTrusts":        adAnalysis.CreateDomainTrustListDelegate(graph.DirectionOutbound),
+			"controllers":           adAnalysis.FetchInboundADEntityControllers,
+			"linkedgpos":            adAnalysis.FetchEntityLinkedGPOList,
+			"dcsyncers":             adAnalysis.FetchDCSyncers,
+		}
+	)
+
+	s.handleAdEntityInfoQuery(response, request, entityType, countQueries)
+}
+
+func (s *Resources) GetGPOEntityInfo(response http.ResponseWriter, request *http.Request) {
+	var (
+		entityType   = ad.GPO
+		countQueries = map[string]any{
+			"ous":         adAnalysis.CreateGPOAffectedIntermediariesListDelegate(adAnalysis.SelectGPOContainerCandidateFilter),
+			"computers":   adAnalysis.CreateGPOAffectedIntermediariesListDelegate(adAnalysis.SelectComputersCandidateFilter),
+			"users":       adAnalysis.CreateGPOAffectedIntermediariesListDelegate(adAnalysis.SelectUsersCandidateFilter),
+			"controllers": adAnalysis.FetchInboundADEntityControllers,
+			"tierzero":    adAnalysis.CreateGPOAffectedIntermediariesListDelegate(adAnalysis.SelectGPOTierZeroCandidateFilter),
+		}
+	)
+	s.handleAdEntityInfoQuery(response, request, entityType, countQueries)
+}
+
+func (s *Resources) GetAIACAEntityInfo(response http.ResponseWriter, request *http.Request) {
+	var (
+		entityType   = ad.AIACA
```

Comment:
I'd prefer us not to perform an allocation here for what is already, essentially, a constant. Just pass it directly in?

Same question for all other patterns of `entityType = X`